internal/ir/xds.go

```diff
@@ -466,15 +467,16 @@ func (r RouteDestination) Validate() error {
 	}
 
 	return errs
-
 }
 
 // DestinationSetting holds the settings associated with the destination
 // +kubebuilder:object:generate=true
 type DestinationSetting struct {
 	// Weight associated with this destination.
 	// Note: Weight is not used in TCP/UDP route.
-	Weight    *uint32                `json:"weight,omitempty" yaml:"weight,omitempty"`
+	Weight *uint32 `json:"weight,omitempty" yaml:"weight,omitempty"`
+	// Protocol associated with this destination/port.
+	Protocol  model.AppProtocol      `json:"protocol" yaml:"protocol"`
```

Comment:
can we move `protocol` to the IR lib ?

Reply:
the same type works, into this file

Reply:
also should this be a ptr, since its optional ?

Reply:
you means move `model.AppProtocol` to package ir? it's OK if you insist.
I put it under model, just because of it's common, shouldn't be too close to package IR.

Reply:
> also should this be a ptr, since its optional?

it can be, but right now make TCP as default.